tests/ext/sugar_test.py

```diff
@@ -93,3 +94,35 @@ def test_streaming(self):
 
     def test_exceptions(self):
         run_exception_test(vimap.ext.sugar.imap_ordered)
+
+
+class ImapOrderedChunkedTests(T.TestCase):
+    def test_basic(self):
+        for n in [2, 4, 8, 32, 3200, 32000]:
+            doubled = tuple(vimap.ext.sugar.imap_ordered_chunked(
+                lambda x: 2 * x,
+                range(n),
+                num_workers=8
+            ))
+            T.assert_equal(doubled, tuple(2 * x for x in range(n)))
+
+    def test_chunking(self):
+        """
+        Makes sure we do chunk the data and each process gets a chunk.
```

Comment:
I would add a bit more explanation about how this test works, but it looks good.